Contributing.md Enhancement

chrisdel101 opened the pull request:
This is a redo of #1671 (which was bungled) and a realization of #1608. 

This changes are all made to the `CONTRIBUTING.md` allowing this file to remain in the repo. This is then copied into the `/resources/contributing.md` page (at the bottom).

All the text content here is fair game for edits, so if any language is unsuitable, something is incorrect, or you just don't like something it can be changed or removed. 
The idea here was just to give a more detailed outline of how to contribute, so if the workflow for that is wrong, it can be corrected. 
I left the translation sections in, but with notices of suspension. 
Also looks like I cannot run the `update-external-docs.yml` myself. This needs to be run to ensure that the script is copied over properly, and for the site contributing page to get populated.

Commit:
add contrbuting changes

Commit:
Reset history to match upstream
- apply all branch changes

Commit:
remove unused script file
changes to soften workflow language

chrisdel101 committed:
Update CONTRIBUTING.md

Add gh important tags to top of page

chrisdel101 committed:
Add corrections for CI paths

chrisdel101 committed:
small text changes to contrib

chrisdel101 commented on _includes/header/header-th.html:
```diff
@@ -113,7 +113,7 @@
                             <a href="/{{ page.lang }}/resources/contributing.html">มีส่วนร่วมกับ Express</a>
                           </li>
                           <li>
-                            <a href="/{{ page.lang }}/changelog/4x.html">บันทึกการเปลี่ยนแปลง</a>
```
This was causing error so changed it.

chrisdel101 replied:
This was removed

chrisdel101 commented on _includes/header/header-id.html:
```diff
@@ -122,7 +122,7 @@
                               <a href="/{{ page.lang }}/resources/contributing.html">Contributing to Express</a>
                             </li>
                             <li>
-                              <a href="/{{ page.lang }}/changelog/4x.html">Release Change Log</a>
```
This was causing error so changed it.

chrisdel101 replied:
This was removed

chrisdel101 committed:
Fix typos/spelling

Commit:
Add uz.css file to fix error

chrisdel101 commented on css/uz.css:
Added since it was missing and causing an error in console in Uz lang.
~~These callouts don't exist in the HTML anywhere according my global search - these files might be removed.~~ My search was wrong. They do exist. Adding this makes the error go away.

chrisdel101 replied:
This was removed

bjohansebas commented:
@chrisdel101 let's try to only modify things related to contributing in this PR; I'll review this PR in detail over the weekend.

chrisdel101 commented:
> @chrisdel101 let's try to only modify things related to contributing in this PR; I'll review this PR in detail over the weekend.

Done.

Please run the pipeline `update-external-docs.yml` to check it works.

bjohansebas commented on CONTRIBUTING.md:
```diff
@@ -1,14 +1,109 @@
 ## Contributing translations
 
+#### Important Notice: We have paused all translation contributions. 
+> [!IMPORTANT]
+> We are currently working toward a more streamlined translations workflow. As long as this notice is posted, we will _not_ be accepting any translation submissions. 
```
Considering that this will be reflected on the website, I wonder how it will look there, since it hasn't appeared yet. Maybe you should run the script to preview it.

bjohansebas replied:
![image](https://github.com/user-attachments/assets/06c89d5e-d450-44d8-ba95-f690340eb882)

chrisdel101 replied:
Ahh your right.
I ran the script over and over on my account to test but it doesn't have the commit access to persist the file, so I must have missed this. Thanks!
Will figure out something here and get back. Would like to keep this styling for the GH side, but the website cannot be like this.

chrisdel101 replied:
I've changed it to this in the website file in both sections it occurs. And checked the link at the top of the page that goes here still works.
![Screenshot 2024-11-18 at 9 27 46 PM](https://github.com/user-attachments/assets/57a9852c-f610-45e0-89ff-e39b0ad8276f)

bjohansebas commented on CONTRIBUTING.md:
```diff
@@ -29,27 +124,31 @@ The documentation is translated into these languages:
+To translate expressjs.com into a new language, follow these steps:
 
+1. Clone the [`expressjs.com`](https://github.com/expressjs/expressjs.com) repository.
+2. Create a directory for the language of your choice using its [ISO 639-1 code](http://www.loc.gov/standards/iso639-2/php/code_list.php) as its name.
```
```suggestion
2. Create a directory for the language of your choice using its [ISO 639-1 code](https://www.loc.gov/standards/iso639-2/php/code_list.php) as its name.
```

chrisdel101 replied:
Fixed.

bjohansebas commented on CONTRIBUTING.md:
```diff
@@ -29,27 +124,31 @@ The documentation is translated into these languages:
+7. Create the header, footer, notice, and announcement file for the language in the `_includes/` directory, in the respective directories, and make necessary edits to the contents.
+8. Create the announcement file for the language in the `_includes/` directory.
+9. Make sure to append `/{{ page.lang }}` to all the links within the site.
+10. Update the `CONTRIBUTING.md` and the `.github/workflows/translation.yml` files with the new language. 
```
Adding the link so they know where to include it would be a good idea.
```suggestion
10. Update the [`CONTRIBUTING.md`](https://github.com/expressjs/expressjs.com/blob/gh-pages/CONTRIBUTING.md#contributing-translations) and the `.github/workflows/translation.yml` files with the new language. 
```

chrisdel101 replied:
Fixed

bjohansebas commented on en/resources/contributing.md:
```diff
@@ -374,3 +376,6 @@ We are currently working on a new version of the security model, the most update
 If you have suggestions on how this process could be improved please submit a
 pull request.
 
+# Contributing to Expressjs.com
```
I would suggest adding the id so it doesn't change over time
```suggestion
# Contributing to Expressjs.com
{#website-contributing}
```

chrisdel101 replied:
Can you elaborate on this? You mean put an HTML tag with ID?

This title is ok to change over time and not break anything as long as it stays above the page anchoring LOCAL line below. 
This title itself is used as an id, see line 12 of https://github.com/expressjs/expressjs.com/pull/1683/files/2c43c76dd5c16c915e5ff6274d7861739faf9d08#diff-4d54134218ddd8f3f2eed0865238cfe6169fbe2007a2256a93935548a747d79a

Let me know if you meant something else.

chrisdel101 replied:
Oh I didn't know you could do this, and it doesn't seem to work in markdown proper, but only when rendered through to HTML. Since the header auto generates an id as shown in that link above, by changing it to `# Contributing to Expressjs.com {#expressjs-website-contributing}` it replaces the ID with the ID written in the braces. The ID needed to be changed at the top too. I made the change `# Contributing to Expressjs.com {#expressjs-website-contributing}`

Is this what you were looking for? Was it just to make it more clear that this section was for website contributing?

bjohansebas replied:
Yeah, that was it.

chrisdel101 replied:
I added this `# Contributing to Expressjs.com {#expressjs-website-contributing}`. It changes the id to that on the website, also had to change the link at the top of the page too.

bjohansebas reviewed (commented):
The work looks great

bjohansebas committed:
run script

chrisdel101 committed:
#1683 request changes work

chrisdel101 commented on en/resources/contributing.md:
```diff
@@ -5,10 +5,14 @@ description: Find out how to contribute to Express.js, including guidelines for
 menu: resources
 lang: en
 redirect_from: "/resources/community.html"
+title2: Contributing to Expressjs.com 
```
i want to remove these two lines. they are not used.

chrisdel101 commented on en/resources/contributing.md:
```diff
@@ -374,3 +378,145 @@ We are currently working on a new version of the security model, the most update
 If you have suggestions on how this process could be improved please submit a
 pull request.
 
+# Contributing to Expressjs.com {#expressjs-website-contributing}
```
i want to add a ‘—-‘ line here to seperate this section.

bjohansebas replied:
If it's fine, it's fine by me.

chrisdel101 replied:
Added
![image](https://github.com/user-attachments/assets/65a77bcc-3636-4a3a-b8e1-1e43b8ef6487)

chrisdel101 reviewed (commented):
@bjohansebas Were these (files are now missing- I'm referring to the commits `ci: update...`) part of this PR? I just reversed them since I thought they were mistakes, then looked again.
I guess you made these on this branch right?
I'll add them back (remove last two commits) in this was your intention.

EDIT: I added these back in. 

EDIT2: Removing them was correct, so they are removed again.

bjohansebas commented:
btw, it's not necessary to run the workflow, running the script and making the commit in this PR is fine. The workflow is just a step to keep the rest of the documentation up to date.

chrisdel101 commented on en/resources/contributing.md:
```diff
@@ -9,6 +9,8 @@ redirect_from: "/resources/community.html"
 # Contributing to Express
 
+### Looking to contribute to expressjs.com? Click [here](#expressjs-website-contributing).
+
```
On line 269 it has this. I was think we should filter this out, or point it to the section below. 
```
### Website Issues
Open issues for the expressjs.com website in https://github.com/expressjs/expressjs.com.
```
![image](https://github.com/user-attachments/assets/c506a9cb-90ef-486e-b1a7-e465cf53938c)

This will be my final change to this PR. 
(Commenting was not allowed on the proper line in this file)

bjohansebas replied:
Maybe it’s better to change it from the source point once this PR is merged.

chrisdel101 replied:
Except they will want to leave this on the Express repo as it is written there, but it doesn't make sense to have it on our page. 
It's a very small change though so we can leave it out if you'd prefer.

chrisdel101 commented on CONTRIBUTING.md:
```diff
@@ -1,14 +1,110 @@
+### The Official Documentation of the Express Framework
 
-Feel free to make changes to the template files or the document files. The supporting docs are located in their respective directories, and the API docs are located under the `_includes` directory.
+>[!NOTE]
```
Okay I made this into a GH tag and it gets filtered out for the site. 
I won't make any more changes other than from reviewers

bjohansebas replied:
sure

bjohansebas commented on .github/workflows/ci.yml:
This shouldn't be part of the PR, maybe you can do something to make it not appear, or if not, just leave it like this.

chrisdel101 replied:
> @bjohansebas Were these (files are now missing- I'm referring to the commits `ci: update...`) part of this PR? 

So you mean the two commits to the CI? I'm not sure if you made those on this branch or they got pulled in by accident. I removed them and then added them back in.
Or put another way, should there be changes to the workflow files here?
If it's both I will remove. Let me know.

chrisdel101 replied:
If you did not make these on this branch, I will def remove them. I was just thinking you made them here for convenience. It was tricky to tell exactly where they came from.

bjohansebas replied:
Those commits are already in the repository, [see](https://github.com/expressjs/expressjs.com/commits/gh-pages/?since=2024-11-16&until=2024-11-16). I don't remember making any commits in this PR apart from running the script last time. Just make changes to the workflow if necessary, but I don't see it as necessary.

chrisdel101 replied:
Doh! Okay then I should have left them removed. I removed them again. Thanks for heads up.

chrisdel101 commented:
@crandmck @wesleytodd @UlisesGascon Sebastian helped with all the technical stuff but could one of you read over the text content and let me know if you want any changes?
The only file you need to read is `CONTRIBUTING.md`

Commit:
remove front matter
    - remove over use of important
    - remove uz.css as it came back
    - removing CI commits
- add HR line
-Add GH NOTE tag and filter out

Commit:
Merge remote-tracking branch 'upstream/gh-pages' into contributing

chrisdel101 commented:
> btw, it's not necessary to run the workflow, running the script and making the commit in this PR is fine. The workflow is just a step to keep the rest of the documentation up to date.

You mean the CI? How do I stop it from running? 
Do you know why the translations fails for me? I get `message: 'Resource not accessible by integration',`

I ran the workflow action outisde this repo to make sure the script paths didn't fail, if this is what you meant.

bjohansebas commented:
> Do you know why the translations fails for me? I get message: 'Resource not accessible by integration' 

This comment explains it quite well (https://github.com/expressjs/expressjs.com/issues/1553#issuecomment-2459914030)

ctcpip commented on CONTRIBUTING.md:
```diff
@@ -1,14 +1,110 @@
+If you want to learn about working on Expressjs.com, this is the right place. Follow the steps below to get started.
+
+#### TL;DR
+1. Open an issue and get approval.
```
is approval necessary?  who is delegated the authority to make these approvals?

wesleytodd replied:
This should not be in here at all imo. The process as with any other repo in this org, is "if it is complicated and/or you dont want to waste your time on a PR that my not land, ask if it is a good idea, otherwise open a PR"

wesleytodd replied:
I am going to copy over a quote from the discussion on this [from the slack](https://openjs-foundation.slack.com/archives/C02QB1731FH/p1732213459272009).

>I think this all makes sense, and thanks for sharing the context it for sure helps. I would guess @crandmck 
 probably is on the same page (correct me if I am wrong) though that approval is not required but opening an issue to discuss is just helpful if you are new or don’t want to spend time opening a PR that might not land.

> This is an OSS lesson I think all of us learned at one point or another, and I would LOVE to improve documentation on this to help others learn without the morale hit or wasted effort. But I just want to be clear, the language in the PR says it is required to get approval, which is not at all the case.

chrisdel101 replied:
I tried to remove any offensive language around this, and can change the wording again, but this idea was literally the point of redoing this document.  Otherwise we might be good to just leave things as they are currently.
This was mentioned before I started this https://github.com/expressjs/expressjs.com/issues/1506#issuecomment-2308979766. 
@bjohansebas Did we not want this step in our process? In my observations, we have been following this process for the past few months at least. 
Approval is informal and can be done by any member.

chrisdel101 replied:
Here is another instance where we used this. https://github.com/expressjs/expressjs.com/pull/1568#issuecomment-2325137955 which resulted in https://github.com/expressjs/expressjs.com/issues/1590
Can we reword this idea somehow to make it more acceptable? Something like:
`If you have any questions or concerns about the utility of your submission, consider opening an issue first to discuss with the team`, etc

crandmck replied:
How about something like this: 
"You're strongly encouraged to open a discussion issue before starting work on a PR, particularly for significant changes that have wide impact.  Small minor improvements or corrections don't need an issue, but for more pervasive contributions, an issue helps to clarify and focus the work and ensure it aligns with the overall project priorities."

chrisdel101 replied:
I've made some changes to the wording in those sections and removed the TL;DR part. 
I tried to soften it, make it seem optional, remove all references to "approval", and and say it's for large changes only.

crandmck commented:
I commented inline, but we should be sure to clarify that no approval is _necessary_ (i.e. in an issue) to start a PR.  IMO an initial discussion issue is strongly recommended for very large contributions, just to ensure they make sense.

The linked discussions came about because of very large refactoring PRs that had no previous discussion at all.  While such PRs are not "prohibited", they can often lead to wasted time not only for the contributor but for the TC/approvers.   External contributors may not always be aware of ongoing discussions or work and can open PRs that don't align with that and are very unlikely to land.   Such PRs are certainly not prohibited, but are not a good use of anyone's time.

Commit:
Make requested lang changes
- remove all "approval" language
- add optional wording
- make everything Express JS wording
- remove italics as it not required
- remove TLDR

chrisdel101 committed:
Update CONTRIBUTING.md

Small typo fixes

bjohansebas commented:
cc: @expressjs/express-tc @expressjs/docs-wg

wesleytodd reviewed (approved):
Awesome! I like the new way we are phrasing the discussion step. This looks great, thanks for the hard work @chrisdel101 and dealing with our "nitpicks" 🤣

UlisesGascon reviewed (approved):
Great work @chrisdel101! :heart:

bjohansebas reviewed (approved):
Awesome

chrisdel101 commented:
i know this review might have taken extra work to get it sounding right so my thanks to everyone for helping to review this one.

bjohansebas commented:
Maybe I'll have to wait for @crandmck approval, since it won't let me merge until the required changes review is removed.

crandmck commented on CONTRIBUTING.md:
```diff
@@ -1,14 +1,109 @@
+
+Whether you're a first time first time contributor or you're just looking for a refresher, visit our [Getting Started](#contributors-guide-to-getting-started) guide below to get up and running.
+
+#### Want to help but need some ideas? These are some of our most typical issue types:
```
The reader is already here because they want to help. Change heading to simply "Need some ideas?" and move "These are some of our most typical issue types:" to be intro sentence.

chrisdel101 replied:
There wasn't any where to put "These are some..."  as an intro sentence so I left it in, but removed the block before that.

crandmck commented on CONTRIBUTING.md:
```diff
@@ -1,14 +1,109 @@
+1.  **Website Related**: 
+If you see anything on the site that could use a tune-up, think about how to fix it.
+
+    - display or screen sizing problems
```
Begin bullet-list items with a capital letter.

crandmck commented on CONTRIBUTING.md:
```diff
@@ -1,14 +1,109 @@
+
+#### Want to help but need some ideas? These are some of our most typical issue types:
+
+1.  **Website Related**: 
```
Sentence case, and rather than "related" I suggest "issues," e.g. "Website issues"

crandmck commented on CONTRIBUTING.md:
```diff
@@ -1,14 +1,109 @@
+    - page structure or user interface enhancements
+
+
+2. **Content Related**: spelling errors, incorrect/outdated Express JS documentation, adding missing content.
```
Same comment as above.

crandmck commented on CONTRIBUTING.md:
```diff
@@ -1,14 +1,109 @@
+    - We love any reports of typos. If you see one, fixing it is a great way to help.
+
+
+3. **Translation Related**: spelling errors, incorrect/poorly translated words, adding new full site translations.
```
Same comment as above.

crandmck commented on CONTRIBUTING.md:
```diff
@@ -1,14 +1,109 @@
+- Submit a [new issue](https://github.com/expressjs/expressjs.com/issues/new/choose) on our repo. Do this for larger proposals, or if you'd like to discuss or get feedback first. 
+- Make a [Github pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request). If you have already done work and it's ready to go, feel free to send it our way.
+
+## Contributor's Guide to Getting Started
```
This wording is confusing, since there's a "Getting started" section of the docs.... Is this about contributing to that? No--so change to just "Getting started".

crandmck commented on CONTRIBUTING.md:
```diff
@@ -1,14 +1,109 @@
+
+The steps below will guide you through the expressjs.com contribution process.
+
+#### Step1: (OPTIONAL) Opening a New Issue
```
Add space: "Step 1"

crandmck commented on CONTRIBUTING.md:
```diff
@@ -1,14 +1,109 @@
+2. After receiving your issue the Express JS documentation team will respond with feedback. We read every submission and always try to respond quickly with feedback. 
+    - For submissions proposing significant change, we encourage you to follow the review process before starting work. 
+
+#### Step2: Get the Application Code Base
```
Add space "Step 2".

crandmck commented on CONTRIBUTING.md:
```diff
@@ -1,14 +1,109 @@
+
+The Express JS website is build using [Jeykyll](https://jekyllrb.com/) and is hosted on [Github Pages](https://pages.github.com/).
+
+#### Step3: Running the Application
```
Add space: "Step 3".

crandmck commented on CONTRIBUTING.md:
```diff
@@ -29,27 +124,31 @@ The documentation is translated into these languages:
-7. Create the announcement file for the language in the `_includes/` directory.
-9. Make sure to append `/{{ page.lang }}` to all the links within the site.
-10. Update the `CONTRIBUTING.md` and the `.github/workflows/translation.yml` files with the new language
+## We're the Express JS Framework's Documentation Team - Not the Express JS Framework
```
This is out of place at the very end of the document.  It doesn't do any good here.  It should be combined with the existing info at the very beginning, where it will do some good... Most of the info is already there already anyway so I think this could just be removed.

chrisdel101 replied:
Awesome feedback @crandmck! Will make the changes.

crandmck reviewed (changes requested):
The content looks good. My comments are mostly copy-edits.  While these aren't critical, since this page is specifically about doc contributions, it should be reasonably exemplary.

I made a number of inline comments.  Also, all the headings need to be revised to be in sentence case, per the [documentation style guide](https://github.com/expressjs/expressjs.com/wiki/Documentation-style-guide#headings).

Commit:
Make requested changes to contrib
- make additional copy edits found during edit

Commit:
More copy changes. Fix sed missing char

chrisdel101 commented:
Hi @crandmck. I made all the changes and actually removed quite a bit of other superfluous language. Can you please review again?
@bjohansebas After this is good to go can you please run the action script again? It's been a long while since it was verified  with many changes since then. It actually would not run locally so I had to add a semicolon after the final `d`, and that was not required before, so hoping it still works. I can also remind you when it is time to do this.
`    sed -E '/^>\[!NOTE\]*/{N;d;}' | \
`
Also looks like some unwanted files in my git history which I need remove still. Just noticed this.

bjohansebas commented:
The script works fine locally, and I don't see anything unusual. Once this is merged, I will run the script on GitHub.

chrisdel101 commented:
So I keep getting unwanted changes in my commits. It happens when I merge I think. This has happened again here and it has ruined my commit history. I don't see anyone else complaining about this so not sure why it happens to me so much but it's happened again here. 

Anyway, the script isn't working correctly now which I'll need to fix. Might be a result of the `git reset`s I had to do. 

The copy is hopefully still okay and can still be reviewed in the meantime.

bjohansebas commented:
I think it was a good idea to enable them to update their branch from the GitHub interface. 

@chrisdel101 I believe I can fix your history now since I have the necessary permissions to do so.

bjohansebas committed:
Small text changes

bjohansebas commented:
@chrisdel101 I think everything is good now, right?

bjohansebas committed:
Merge branch 'gh-pages' into contributing

chrisdel101 commented:
> @chrisdel101 I think everything is good now, right?

Yes it looks okay. Thanks for fixing that, however you did it! I've been pulling into my local branch but this has consistently led to problems. What's the best way to merge?
Can it be merged on github before the PR is accepted? I think I have permissions to do that now.

bjohansebas commented:
> What's the best way to merge?

![image](https://github.com/user-attachments/assets/8fbb9b34-0c43-4586-9ea9-946046356794)
You can use this new button on GitHub that I activated, so you can get the latest changes from the upstream main branch.
> Can it be merged on github before the PR is accepted?

@chrisdel101 No, you always have to wait until there are no required changes (especially when it's a team member) and I have approval from someone with write permissions (in some cases, it will be necessary for a TC member to approve it as well). If you want, you can contact @crandmck on Slack to get his approval if he wants. Once there are no more required changes and we all agree, you can merge this.

Commit:
Fix script to re-write on each run
- remove {{content}} as it renders in liquid

chrisdel101 commented:
Alrighty I fixed the script file and it's working as expected. I tested it on my action and it ran. It didn't like the lines like these `(#expressjs-website-contributing)` and `{#expressjs-website-contributing}`, so needed to add logic to handle those.
Hoping we can get this one put away soon :) @crandmck could you check out your requested changes? 
Btw you can see the CONTRIBUTING.md [here](https://github.com/chrisdel101/expressjs.com/blob/contributing/CONTRIBUTING.md)

Commit:
clarify comments - remove confusing -> symbols

crandmck reviewed (approved):
Thanks for sticking with it and seeing this one through!

crandmck committed:
Merge branch 'gh-pages' into contributing
